lerobot/scripts/train.py

```diff
@@ -248,13 +253,58 @@ def train(cfg: DictConfig, out_dir: str | None = None, job_name: str | None = No
     set_global_seed(cfg.seed)
 
     logging.info("make_dataset")
-    offline_dataset = make_dataset(cfg)
+    is_multi_dataset = isinstance(cfg.dataset_repo_id, ListConfig)
+    if is_multi_dataset:
+        # If multiple dataset repo IDs are provided we make multiple corresponding LeRobot datasets and
+        # concatentate them.
+        _offline_datasets = [make_dataset(cfg, dataset_repo_id=repo_id) for repo_id in cfg.dataset_repo_id]
+        # Disable any data keys that are not common across all of the datasets.
+        intersection_data_keys = set(_offline_datasets[0].hf_dataset.features)
+        for dataset in _offline_datasets:
+            intersection_data_keys.intersection_update(dataset.hf_dataset.features)
+        if len(intersection_data_keys) == 0:
+            raise RuntimeError(
+                "Multiple datasets were provided but they had no keys common to all of them. The "
+                "multi-dataset functionality currently only keeps common keys."
+            )
+        for dataset_repo_id, dataset in zip(cfg.dataset_repo_id, _offline_datasets, strict=True):
+            extra_keys = set(dataset.hf_dataset.features).difference(intersection_data_keys)
+            logging.warning(
+                f"keys {extra_keys} of {dataset_repo_id} were disabled as they are not contained in all the "
+                "other datasets."
+            )
+            dataset.disable_data_keys(extra_keys)
+        # Add a transform that gives each dataset an index (corresponding to the order in which the  datasets
+        # were provided in the config).
+        id_to_index = {repo_id: i for i, repo_id in enumerate(cfg.dataset_repo_id)}
+        logging.info(
+            "Multiple datasets were provided. Applying the following index mapping to the provided datasets: "
+            f"{pformat(id_to_index, indent=2)}"
+        )
+        for dataset_repo_id, dataset in zip(cfg.dataset_repo_id, _offline_datasets, strict=True):
+
+            def transform(item, d_id=dataset_repo_id):
+                item["dataset_index"] = torch.tensor(id_to_index[d_id])
+                return item
```

Comment:
We should probably avoid using transform for this use case. That's why I would prefer to use a `MultiLeRobotDataset` class that would contain the mapping in its attribute and add the `dataset_index` key in its getitem.
What do you think?

Reply:
Agreed. I will wrap up your comments and start a separate PR with the `MultiLeRobotDatset` approach.